tests/test_hydra.py

```diff
@@ -730,6 +730,24 @@ def test_local_run_workdir(
     )
 
 
+@pytest.mark.parametrize(
+    "task_config",
+    [
+        ({"hydra": {"run": {"dir": "${now:%Y%m%d_%H%M%S_%f}"}}}),
+    ],
+)
+def test_run_dir_microseconds(tmpdir: Path, task_config: DictConfig) -> None:
+    cfg = OmegaConf.create(task_config)
+    assert isinstance(cfg, DictConfig)
+    integration_test(
+        tmpdir=tmpdir,
+        task_config=cfg,
+        overrides=[],
+        prints="str('%f' not in os.getcwd())",
```

Comment:
curious why you need str() here.

Reply:
print() function appends \n, and the expression becomes str('\f' not in os.getcwd())+ "\n". Without str(), the expression fails.

Reply:
File "/home/xxx/usr/venv-3.9/lib/python3.9/site-packages/hydra/core/utils.py", line 129, in run_job
    ret.return_value = task_function(task_cfg)
  File "/tmp/pytest-of-xxx/pytest-7/test_run_dir_microseconds_task0/task.py", line 10, in experiment
    f.write('%f' not in os.getcwd() + "\n")

Reply:
> TypeError: write() argument must be str, not bool

Okay, so this is the real reason.